### PR TITLE
Added GMS-like implementation for 'Regular Cab in Victoria' NPC

### PR DESCRIPTION
**Pull request by Silwhoon**
Includes
- GMS-like text
- Meso reduction in player chat box
- Dynamic destination options based on the players current map

### CODE REVIEW

**Comment by Silwhoon** [comment]
@doriyan13 If there is ANY feedback big or small please let me know

File: src/main/java/com/dori/SpringStory/client/character/MapleChar.java [hunk 1]
```diff
@@ -941,7 +941,7 @@ public Drop dropItem(Item item) {
         return dropItem(item, 0);
     }
 
-    public void modifyMeso(int amount) {
+    public void modifyMeso(int amount, boolean showInChat) {
```
Comment (doriyan13) [inline src/main/java/com/dori/SpringStory/client/character/MapleChar.java:944]: if you modify an already existing method that had set amount of params, i suggest take a advantage of overloading -

modify the base method as you did but add a new method that keep the original logic with default value of true so - 

`public void modifyMeso(int amount, boolean showInChat) {
  modifyMeso(amount, true);
}
`

File: src/main/java/com/dori/SpringStory/client/character/MapleChar.java [hunk 2]
```diff
@@ -952,6 +952,8 @@ public void modifyMeso(int amount) {
             setMeso(newMoneyAmount);
             updateStat(Money, newMoneyAmount);
         }
+
+        if (showInChat) write(CWvsContext.incMoneyMessage(amount));
```
Comment (doriyan13) [inline src/main/java/com/dori/SpringStory/client/character/MapleChar.java:956]: in java is preferred to use curly braclets so add `{ }`

File: src/main/java/com/dori/SpringStory/scripts/npcs/Common.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        List<Integer> taxiMaps = Arrays.asList(100000000, 101000000, 102000000, 103000000, 104000000, 105000000, 120000000);
+        List<MenuOption> menuOptions = new ArrayList<>();
+        for (Integer mapId : taxiMaps) {
+            if (chr.getMapId() != mapId) {
+                // Once big bang hit, all taxi options cost 1,000 meso
+                menuOptions.add(NpcScriptUtils.addTaxiMoveOption(script, chr, mapId, chr.getJob() == 0, 1000));
+            }
+        }
+
+        script.askMenu("Please select your destination.\r\n", menuOptions.toArray(MenuOption[]::new));
```
Comment (doriyan13) [inline src/main/java/com/dori/SpringStory/scripts/npcs/Common.java:37]: it's wastful memory wise, but i do get why you did it cuz i didn't support list option, i will fix it soon :D

Comment (Silwhoon) [reply src/main/java/com/dori/SpringStory/scripts/npcs/Common.java:37]: Is there a better way of achieving this right now? Or should I change it once you've made your change

Comment (doriyan13) [reply src/main/java/com/dori/SpringStory/scripts/npcs/Common.java:37]: change once i add support to that i think :D

**Review by doriyan13 — APPROVED** [review]
First contribution 🥳

**Pull request merged**